src/tools/pbs_upgrade_job.c

```diff
@@ -289,36 +200,16 @@ check_job_file(int fd)
 	int ret_version = -1;
 	int length = -1;
 	jobfix_PRE19 old_jobfix_pre19;
+
 	/* Save our current position so we can comeback to it */
 	pos_saved = lseek(fd, 0, SEEK_CUR);
 
-	/*---------- For PBSPro 12 version ---------- */ 
-	/* The string "Job_Name" should appear at address 01160 (octal) for v12 */
-	pos_new = lseek(fd, 01160, SEEK_SET);
-	if (pos_new != 01160) {
-		fprintf(stderr, "File format not recognized.\n");
-		goto check_job_file_exit;
-	}
-	errno = 0;
-	length = read(fd, &buf, 8);
-	if (length < 0) {
-		fprintf(stderr, "Failed to read input file [%s]\n",
-				errno ? strerror(errno) : "No error");
-		goto check_job_file_exit;
-	}
-	if (length != 8) {
-		fprintf(stderr, "Format not recognized, no attributes.\n");
-		goto check_job_file_exit;
-	}
-	buf[8] = '\0';
-	if (strcmp(buf, ATTR_name) == 0) {
-		ret_version = 12;
-		goto check_job_file_exit;
-	}
-
-	/*---------- For PBSPro >=13 or <=18 versions job structure ---------- */
-	length = -1;
+	/*---------- For PBSPro >=13.x or <=18.x versions jobfix structure ---------- */
 	pos_new = lseek(fd, pos_saved, SEEK_SET);
+	if (pos_new != 0) {
+			fprintf(stderr, "Couldn't set the file position to zero.\n");
```

Comment:
Please print errno also as part of the message which tells us why lseek failed.

Reply:
fixed!